cdc/processor/pipeline/sink.go

```diff
@@ -315,17 +322,47 @@ func (n *sinkNode) Receive(ctx pipeline.NodeContext) error {
 			failpoint.Inject("ProcessorSyncResolvedError", func() {
 				failpoint.Return(errors.New("processor sync resolved injected error"))
 			})
-			if err := n.flushSink(ctx, msg.PolymorphicEvent.CRTs); err != nil {
+
+			err := retry.Do(ctx, func() error {
+				err := n.flushSink(ctx, msg.PolymorphicEvent.CRTs)
+				if err != nil {
+					return err
+				}
+				return nil
+			}, retry.WithBackoffBaseDelay(20),
+				retry.WithInfiniteTries(),
+				retry.WithIsRetryableErr(isRetryableError))
+			if err != nil {
 				return errors.Trace(err)
 			}
+
 			atomic.StoreUint64(&n.resolvedTs, msg.PolymorphicEvent.CRTs)
 			return nil
 		}
-		if err := n.emitEvent(ctx, event); err != nil {
+
+		err := retry.Do(ctx, func() error {
+			err := n.emitEvent(ctx, event)
+			if err != nil {
+				return err
+			}
+			return nil
+		}, retry.WithBackoffBaseDelay(20),
+			retry.WithInfiniteTries(),
+			retry.WithIsRetryableErr(isRetryableError))
+		if err != nil {
 			return errors.Trace(err)
 		}
 	case pipeline.MessageTypeTick:
-		if err := n.flushSink(ctx, n.resolvedTs); err != nil {
+		err := retry.Do(ctx, func() error {
```

Comment:
Maybe we can wrap the retry logic in a helper function?

Reply:
got